src/components/Chat/BottomSheetGif.js

```diff
@@ -48,15 +41,33 @@ const BottomSheetGif = React.forwardRef((props, ref) => {
     setIsLoading(false);
   };
 
+  const searchGIF = React.useCallback(
+    debounce((search) => {
+      onLoad(search);
+    }, 500),
+    []
+  );
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+    if (text?.trim()?.length > 2) {
+      searchGIF(text);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The creation of the `searchGIF` function using `useCallback` and `debounce` is a good practice to optimize the search functionality. It prevents the `onLoad` function from being called excessively when the user is typing in the search box. The condition in `handleChangeText` that checks if the text length exceeds 2 before calling `searchGIF` is also a good practice to avoid unnecessary API calls for too short search queries.
```diff
+ const searchGIF = React.useCallback(
+   debounce((search) => {
+     onLoad(search);
+   }, 500),
+   []
+ );
+
+ const handleChangeText = (text) => {
+   setSearchText(text);
+   if (text?.trim()?.length > 2) {
+     searchGIF(text);
+   }
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->